primitives/version/src/lib.rs

```diff
@@ -207,6 +235,49 @@ pub struct RuntimeVersion {
 	///
 	/// It need *not* change when a new module is added or when a dispatchable is added.
 	pub transaction_version: u32,
+
+	/// Version of the state implementation used by this runtime.
+	/// Use of an incorrect version is consensus breaking.
+	pub state_version: u8,
+}
+
+impl RuntimeVersion {
+	/// `Decode` implementation, with
+	/// possibility to inject the apis
+	/// version (when stored at a different
+	/// location).
+	pub fn decode_with_apis<I: Input>(
+		input: &mut I,
+		with_apis: Option<ApisVec>,
+	) -> Result<RuntimeVersion, codec::Error> {
+		let spec_name = Decode::decode(input)?;
+		let impl_name = Decode::decode(input)?;
+		let authoring_version = Decode::decode(input)?;
+		let spec_version = Decode::decode(input)?;
+		let impl_version = Decode::decode(input)?;
+		let apis = if let Some(apis) = with_apis { apis } else { Decode::decode(input)? };
```

Comment:
This is wrong. The version in the wasm section still contains an api object. Which is just "0", aka an empty vec.

Could you please write the function "decode_with_version_hint` as I said before.

Reply:
ah ok (please forget the last commit)